library-extension-rx/src/main/java/com/instacart/library/truetime/TrueTimeRx.java

```diff
@@ -33,8 +33,19 @@ public static TrueTimeRx build() {
         return RX_INSTANCE;
     }
 
-    public TrueTimeRx withSharedPreferences(Context context) {
-        super.withSharedPreferences(context);
+    public TrueTimeRx withSharedPreferencesCache(Context context) {
+        super.withSharedPreferencesCache(context);
+        return this;
+    }
+
+    /**
+     * Provide your own cache interface to cache the true time information.
+     * Please be noted that if you provide such cache interface, it is also your own responsibility
+     * to clear the cache on device reboot. This is a must.
```

Comment:
we have a [broadcast receiver](https://github.com/instacart/truetime-android/blob/master/library/src/main/java/com/instacart/library/truetime/BootCompletedBroadcastReceiver.java) that calls `TrueTime.clearCachedInfo(context);` on a boot. If we then delegate that to the CacheInterface#clear, that's not foolproof ?

Reply:
I had put a second thought about it. And it makes sense for us to use the build-in Broadcast Receiver. It should be noted that the library users must initialize the TrueTime in `onCreate` in `Application.class` rather than in the activity.  The reason I initially added this comment is because when I was integrating TrueTime library into my app, I actually did not initialize TrueTime in `onCreate` in `Application.class`, I initialized it in one of my activity, which means that our `_cacheInterface` was not setup when the device is rebooted and failed in clearing the cache.